src/build-extensions-container.sh

```diff
@@ -18,10 +18,16 @@ if [ -d "${workdir}/src/yumrepos" ]; then
     find "${workdir}/src/yumrepos/" -maxdepth 1 -type f -name '*.repo' -exec cp -t "${ctx_dir}" {} +
 fi
 
-# Build the image, replacing the FROM directive with the local image we have
+variant=""
+if [[ -f "${workdir}/src/config.json" ]]; then
+    variant="$(jq --raw-output '."coreos-assembler.config-variant"' "${workdir}/src/config.json")"
+fi
```

Comment:
Shouldn't this be using the variant info from the build we're extending instead?

I think it's confusing to have both `src/config.json` and the variant in the build metadata. Can we delete the latter and just make it a `cosa build` switch instead? It only matters for the initial build anyway since everything else should be operating on an existing build, in which case we'd want it to care about the variant of the build itself.

Reply:
OK, now I understand the idea. That would indeed likely simplify things and remove the `config.json` file.

But this requires changes in CI scripts and several places in COSA so not a simple change.

Reply:
Right, let's not block on this for https://github.com/coreos/fedora-coreos-pipeline/pull/787, but I think it'd be a nice conceptual cleanup to do soon-ish before more things try to consume `src/config.json`.

Reply:
But this also means that we change the semantics of what we do in that we'll always have to pass the variant option to `cosa build` instead of it being fixed at repo init time.

Reply:
You'll have to pay the cost of "manually specify variant" at least once regardless. This would move it from `cosa init` time to `cosa build` time. We could have `cosa build` default to the variant it last built if there's a previous build locally so that in the dev path, it doesn't have to be repeated each time.

Reply:
OK works for me

Reply:
OK, see https://github.com/coreos/coreos-assembler/issues/3276 for a proposal with pending questions.